hippy/frame.py

```diff
@@ -27,9 +30,33 @@ def match(self, interp, w_exc):
             return False
         return k_exc.is_subclass_of_class_or_intf_name(k_catch.name)
 
+    def py_match(self, frame, w_exc):
+        'Try to match w_exc to a Python exception.'
+        if not isinstance(w_exc, W_PyExceptionAdapter):
+            return False
+
+        interp = frame.interp
+        # try python builtin if there is no py parent
+        if frame.bytecode.py_scope is None:
+            try:
+                w_py_cls = interp.py_space.builtin.get(self.exc_class)
+            except OperationError:
```

Comment:
you should not catch any OperationError here. you have to match that it's the right class inside (a space.w_KeyError?). Otherwise things like KeyboardInterrupts could get ignored here, which are also OperationErrors.